core/src/mindustry/entities/UnitSorts.java

```diff
@@ -11,7 +11,25 @@ public class UnitSorts{
     closest = Unit::dst2,
     farthest = (u, x, y) -> -u.dst2(x, y),
     strongest = (u, x, y) -> -u.maxHealth + Mathf.dst2(u.x, u.y, x, y) / 6400f,
-    weakest = (u, x, y) -> u.maxHealth + Mathf.dst2(u.x, u.y, x, y) / 6400f;
+    weakest = (u, x, y) -> u.maxHealth + Mathf.dst2(u.x, u.y, x, y) / 6400f,
+    //Fires at roughly the center of a unit cluster.
+    grouped = (u, x, y) -> {
```

Comment:
The performance of this is not acceptable:
- Every unit will iterate through every other unit within a large radius: O(N^2) complexity, very bad.
- Multiple allocations when targeting (use a static external variable at least)